### ISSUE TITLE
TSM list for 0.9.6

### ISSUE THREAD

**Issue by pauldix** [influxdb/influxdb#4976]
The list of remaining items to have TSM ready for testing for the 0.9.6 release on December 8th. There will be another issue for work to be completed after the 8th.

If you're taking an item, put your name at the end of it.

### Functionality
- [x] Cache query should support time-descending ordering. (@otoolep) https://github.com/influxdb/influxdb/pull/4980
- [x] Update Cache cursor to combine snapshot caches and the hot cache (@otoolep)
- [x] Re-work TSM compactions w/ updated design (@jwilder)
- [x] Cleanup of tmp files from failed compactions (@jwilder)
- [x] Make `cache.Values` goroutine safe
- [x] Delete series/measurement (bonus points for making this flexible so that it'll work for `DELETE FROM` statements that give specific time ranges) (@jwilder)
- [x] Load Database Index from items only in WAL at startup (@pauldix)
- [x] Corrupted WAL segment handling (@otoolep) https://github.com/influxdb/influxdb/pull/5006
- [x] Handle delete WAL entries when loading cache at startup (@pauldix)
- [x] Update the Plan logic to have some tunable parameters around when TSM files are selected for compaction. For instance, only compacting files older than some given time. Then compacting full after no writes have been received for a while (@pauldix)
- [x] Add some logic for forcing a snapshot write if a shard hasn't received a write in some configurable amount of time (@pauldix)
- [x] Config options (add/remove/cleanup) (@pauldix)
- [x] Change Dev to the new TSM1 (@pauldix)
- [x] Remove MinTime from encoded blocks (tsm1 replacement must happen first) (@jwilder)

cc @jwilder, @otoolep

**Comment by otoolep** [comment]
https://github.com/influxdb/influxdb/issues/4976 is list 1 for reference.

**Comment by pauldix** [comment]
Closing this out since we're done here. Let's get #4977 done!
